web/modules/custom/dpl_login/src/OpenIdUserInfoService.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Drupal\dpl_login;
+
+use Drupal\Core\Site\Settings;
+
+use function Safe\sprintf;
+
+/**
+ * Access Token.
+ */
+class OpenIdUserInfoService {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private Settings $settings
+  ) {}
+
+  /**
+   * Get the user info from the response.
+   *
+   * From the adgangsplatformen userinfo endpoint.
+   *
+   * @param mixed[] $response
+   *   The response from the userinfo endpoint.
+   *
+   * @return mixed[]
+   *   The openid_connect user info.
+   */
+  public function getOpenIdUserInfoFromAdgangsplatformenUserInfoResponse(array $response): array {
+    $name = uniqid();
+    // Drupal needs an email. We set a unique one to apply to that rule.
+    $userinfo['email'] = sprintf('%s@dpl-cms.invalid', $name);
+    // Drupal needs a username. We use the unique id to apply to that rule.
+    $userinfo['name'] = $name;
+    // openid_connect module needs the sub for creating the auth map.
+    $userinfo['sub'] = $this->getSubIdFromUserInfo($response);
+
+    return $userinfo;
+  }
+
+  /**
+   * Get the hashed userinfo id.
+   *
+   * @param mixed[] $userinfo
+   *   The userinfo from the adgangsplatformen userinfo endpoint.
+   *
+   * @return string
+   *   The hashed opemid_connect sub id.
+   */
+  public function getSubIdFromUserInfo(array $userinfo): string {
+    $identifier_data = $this->getIdentifierDataFromUserInfo($userinfo);
+    return $this->hashIdentifier($identifier_data['id']);
+  }
+
+  /**
+   * Get the identifier data from the userinfo.
+   *
+   * @param mixed[] $userinfo
+   *   The userinfo from the adgangsplatformen userinfo endpoint.
+   *
+   * @return mixed[]
+   *   The identifier data: Raw id and type of id.
```

Comment:
```suggestion
   * @return array{"id": string, "type": AuthorizationIdType}
   *   The identifier data: Raw id and type of id.
```

We could use array shapes for a little more safety here.